packages/grafana-ui/src/utils/index.ts

```diff
@@ -8,6 +8,7 @@ export { getFlotPairs, getFlotPairsConstant } from './flotPairs';
 export * from './slate';
 export * from './dataLinks';
 export { default as ansicolor } from './ansicolor';
+export { useDelayedSwitch } from './useDelayedSwitch';
```

Comment:
Not sure this should be exposed from grafana-ui. Exports from index files in grafana-ui are available to import from @grafana/ui . I would remove it from here and rely on relative import if it's internal to grafana-ui now.

Reply:
In general I think we should use something like this every time we show some loading state so I thought it makes sense to export it and let people know they should use it.

Reply:
I agree, but I think instead of exposing hook we should expose a set of loading indicator components(that use this hook) to use in different contexts.

Reply:
Yeah that would also make sense but not sure whether that would work well with this particular use case where you want to also have transition between those states and so need the delayed loading state exposed.

Reply:
Will keep it internal for now until there are more usecases.